src/chain/tracker.rs

```diff
@@ -116,32 +123,141 @@ pub fn start_chain_watch(
+                                                            if !paid_db {
+                                                                match invoice.check(&client, &watcher, &block).await {
+                                                                    Ok(paid) => {
+                                                                        if paid {
+                                                                            state.order_paid(id.clone()).await;
+                                                                        }
+                                                                    }
+                                                                    Err(e) => {
+                                                                        tracing::warn!("account fetch error: {0:?}", e);
+                                                                    }
+                                                                }
+
+                                                                checked_accounts.insert(id);
+                                                            }
+
+                                                            tracing::debug!("Removing an account {id:?} due to passing its death timestamp");
+                                                            id_remove_list.push(id.to_owned());
+                                                        }
+                                                        Err(e) => {
+                                                            tracing::warn!("account read error: {e:?}");
+                                                        }
+                                                    }
+                                                }
+                                            }
+                                        }
+                                    }
+                                    Err(e) => {
+                                        tracing::warn!("events fetch error: {0:?}", e);
+                                    }
+                                }
+
                                 // Important! There used to be a significant oprimisation that
                                 // watched events and checked only accounts that have tranfers into
                                 // them in given block; this was found to be unreliable: there are
                                 // ways to transfer funds without emitting a transfer event (one
                                 // notable example is through asset exchange procedure directed
                                 // straight into invoice account), and probably even without any
-                                // reliably expected event (through XCM). Thus we just scan all
-                                // accounts, every time. Please submit a PR or an issue if you
+                                // reliably expected event (through XCM). Thus we just scan almost
```

Comment:
why "almost"?

Reply:
Actually, no, I've optimized this:
https://github.com/fluiderson/kalatori/blob/54d7b24093779a17b918c219feb904acaccabea0/src/chain/tracker.rs#L251C78-L251C126

Reverting the reverted...

Reply:
is this optimization useful? In the end, we fetch all accounts anyway, and it will miss eventless transfers (XCM, payout, thing-that-was-not-invented-yet-but-will-be-tomorrow). Maybe just scan them all for rigidity?

Reply:
Better strategy to check for withdrawal would be checking balance and then backsearch to find event, but since we haven't done this yet, this is good enough

Reply:
https://github.com/Alzymologist/Kalatori-backend/pull/107/commits/dae8333c636edf884a0706ff3245a6122b7175a9